nncf/quantization/algorithms/weight_compression/torch_fx_backend.py

```diff
@@ -0,0 +1,273 @@
+        lora_correction_algo: LoraCorrectionAlgorithm = None,
+    ) -> torch.fx.GraphModule:
+        transformation_layout = TransformationLayout()
+
+        for wc_params in weight_compression_parameters:
+            compression_config = wc_params.compression_config
+            if compression_config.mode not in [
+                CompressWeightsMode.INT8_ASYM,
+                CompressWeightsMode.INT8_SYM,
+                CompressWeightsMode.INT8,
+            ]:
+                raise ValueError(f"{compression_config.mode.value} is not supported.")
+            weight_node = get_const_node(wc_params.node_with_weight, wc_params.weight_port_id, graph)
+            weight_name = weight_node.node_name
+            consumer_nodes = graph.get_next_nodes(weight_node)
+            weight = self.get_weight(wc_params.node_with_weight, wc_params.weight_port_id, model, graph)
+            if weight is None or not isinstance(weight, Tensor):
+                raise nncf.InternalError(f"Could not find a nncf.tensor in the model by name {weight_name}.")
+
+            # calculates compressed weights and decompression parameters
+            compressed_weight = compress_weight(
+                weight,
+                wc_params.reduction_axes,
+                compression_config,
+                None if precomputed_scales is None else precomputed_scales.get(wc_params.weight_name),
+                None if precomputed_zero_points is None else precomputed_zero_points.get(wc_params.weight_name),
+            )
+            compressed_weight.scale = compressed_weight.scale.astype(dtype=TensorDataType.float16)
+
+            # pack compressed tensor
+            if compression_config.mode == CompressWeightsMode.INT8_SYM:
+                dtype = TensorDataType.int8
+            else:
+                dtype = TensorDataType.uint8
+            packed_tensor = compressed_weight.tensor.astype(dtype)
+
+            self.set_weight(wc_params.node_with_weight, wc_params.weight_port_id, model, graph, packed_tensor)
+
+            if len(consumer_nodes) > 1:
+                raise nncf.InternalError("Shared weights not supported in compression for TorchFX models")
+
+            # creates weight decompressor
+            if compression_config.mode == CompressWeightsMode.INT8_SYM:
+                decompressor = SymmetricWeightsDecompressor(
+                    compressed_weight.scale.data, result_dtype=weight.data.dtype
+                )
+                decompressor_type = "symmetric"
+            else:
+                packed_zero_point = compressed_weight.zero_point.astype(dtype)
+                decompressor = AsymmetricWeightsDecompressor(
+                    compressed_weight.scale.data, packed_zero_point.data, result_dtype=weight.data.dtype
+                )
+                decompressor_type = "asymmetric"
+
+            # registry weight decompression module in the model
+            # TODO: Find a more efficient way to access updated constant name
```

Comment:
I would suggest to solve this issue in this PR and remove implicit connection between weight assigment and get weight name.

Reply:
Alright